src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -63,7 +63,7 @@ client_ptr kafka_client_cache::make_client(
       to_yaml(cfg, config::redact_secrets::no));
 }
 
-client_ptr kafka_client_cache::fetch_or_insert_impl(
+std::pair<client_ptr, client_sm_ptr> kafka_client_cache::fetch_or_insert_impl(
```

Comment:
I can't see how splitting the impl out helps at all.

Reply:
You recommended this in slack so the code better shows that `fetch_or_insert` is synchronous.

Reply:
I think I pointed out that `fetch_or_insert` being synchronous simplifies the understanding of it, and that it should remain that way.